### PR TITLE
Clarify that [patch] can also use sources that are not crates.io

### PR DESCRIPTION
**Pull request by federicomenaquintero**
This includes an example of how to use the `[patch.'https://blahblah']` syntax.  I couldn't figure out how to make this work until I saw a crate that actually did this.

### CODE REVIEW

File: src/doc/manifest.md [hunk 1]
```diff
@@ -720,19 +720,28 @@ other copies. The syntax is similar to the `[dependencies]` section:
 [patch.crates-io]
 foo = { git = 'https://github.com/example/foo' }
 bar = { path = 'my/local/bar' }
+
+[dependencies.baz]
+git = 'https://github.com/example/baz'
+
+[patch.'https://github.com/example/baz']
+baz = { git = 'https://github.com/example/patched-baz', branch='my-branch' }
```
Comment (matklad) [inline src/doc/manifest.md]: Micronit: let's add spaces around `=` in `branch='my-branch`, to be consistent with other examples here.

Comment (federicomenaquintero) [reply src/doc/manifest.md]: Oops, of course.  Added them.

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Comment by alexcrichton** [comment]
@federicomenaquintero that failure isn't your fault, we'll work on getting that fix and get this merged.

**Pull request merged**